src/constructs/autoscaling/asg.test.ts

```diff
@@ -295,4 +296,85 @@ describe("The GuAutoScalingGroup", () => {
       },
     });
   });
+
+  test("has an instance role created by default with AssumeRole permissions", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      ...defaultProps,
+      app: "TestApp",
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Role", {
+      AssumeRolePolicyDocument: {
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Action: "sts:AssumeRole",
+            Effect: "Allow",
+            Principal: { Service: { "Fn::Join": ["", ["ec2.", { Ref: "AWS::URLSuffix" }]] } },
+          },
+        ],
+      },
+    });
+  });
+
+  test("passing in an instance role overrides the default", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      ...defaultProps,
+      app: "TestApp",
+      role: new GuInstanceRole(stack, {
+        app: "OverridenAppName",
+      }),
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Role", {
+      Tags: [
+        { Key: "App", Value: "OverridenAppName" },
+        { Key: "gu:cdk:version", Value: "TEST" },
+        { Key: "Stack", Value: "test-stack" },
+        { Key: "Stage", Value: { Ref: "Stage" } },
+      ],
+    });
+  });
+
+  test("has sensible default scaling capacities per stage based on minimum capacity", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      ...defaultProps,
```

Comment:
I think we need to remove some [default props from the test data](https://github.com/guardian/cdk/blob/c061c678305fec65660e0421216d441b962ad8be/src/constructs/autoscaling/asg.test.ts#L26-L32) to confirm that this test is working as expected.

Reply:
Ah! Good catch, thanks. Tests and code have been updated to reflect this.